File: _test_sample.js
const { scorePost, comparePosts } = require("./engagementAlgo");
const { improvePost } = require("./improver");

const SAMPLE = `the new year is almost here

2026 has been one of my most productive years

now I'm planning for the next few months

using 1plann I'm keeping track of each task I complete

https://1plann.space`;

const sep = "=".repeat(70);
console.log(sep);
console.log("ORIGINAL POST");
console.log(sep);
console.log(SAMPLE);
console.log();

const r = scorePost(SAMPLE);
console.log(`SCORE: ${r.score}  GRADE: ${r.grade}`);
console.log(`VERDICT: ${r.verdict}`);
console.log();
console.log("BREAKDOWN:");
for (const b of r.breakdown) {
  console.log(`  ${b.dimension.padEnd(16)} ${b.score.toFixed(1).padStart(5)}  (w=${b.weight.toFixed(2)})  ${b.note}`);
}
console.log();
console.log("PROBLEMS:");
for (const p of r.problems) console.log(`  - ${p}`);
console.log();
console.log("SIGNALS TO ADD:");
for (const s of r.signalsToAdd) console.log(`  + ${s}`);
console.log();

console.log(sep);
console.log("IMPROVING...");
console.log(sep);
const res = improvePost(SAMPLE);
console.log(`\nFINAL SCORE: ${res.finalScore}  GRADE: ${res.finalGrade}  (converged=${res.converged})`);
console.log();
console.log("ITERATIONS:");
for (const step of res.iterations) {
  console.log(`  iter ${step.iteration}: score=${step.score} grade=${step.grade} changes=${JSON.stringify(step.changes)}`);
}
console.log();
console.log("FINAL POST:");
console.log("-".repeat(70));
console.log(res.final);
console.log("-".repeat(70));
if (res.linkReply) {
  console.log("\nSUGGESTED FIRST REPLY:");
  console.log(res.linkReply);
}
console.log("\nTIMING:", res.timingAdvice);

// A/B demo
console.log("\n" + sep);
console.log("A/B COMPARISON: original vs final");
console.log(sep);
const cmp = comparePosts(SAMPLE, res.final);
console.log(cmp.summary);
console.log(`  A (original): ${cmp.a.score} ${cmp.a.grade}`);
console.log(`  B (final):    ${cmp.b.score} ${cmp.b.grade}`);

File: reference/_test_sample.py
"""Quick smoke test against the user's real sample post."""
from engagement_algo import score_post, compare_posts
from improver import improve_post
import json

SAMPLE = """the new year is almost here

2026 has been one of my most productive years

now I'm planning for the next few months

using 1plann I'm keeping track of each task I complete

https://1plann.space"""

print("=" * 70)
print("ORIGINAL POST")
print("=" * 70)
print(SAMPLE)
print()

r = score_post(SAMPLE)
print(f"SCORE: {r.score}  GRADE: {r.grade}")
print(f"VERDICT: {r.verdict}")
print()
print("BREAKDOWN:")
for b in r.breakdown:
    print(f"  {b.dimension:16s} {b.score:5.1f}  (w={b.weight:.2f})  {b.note}")
print()
print("PROBLEMS:")
for p in r.problems:
    print(f"  - {p}")
print()
print("SIGNALS TO ADD:")
for s in r.signals_to_add:
    print(f"  + {s}")
print()

print("=" * 70)
print("IMPROVING...")
print("=" * 70)
res = improve_post(SAMPLE)
print(f"\nFINAL SCORE: {res.final_score}  GRADE: {res.final_grade}  (converged={res.converged})")
print()
print("ITERATIONS:")
for step in res.iterations:
    print(f"  iter {step.iteration}: score={step.score} grade={step.grade} changes={step.changes}")
print()
print("FINAL POST:")
print("-" * 70)
print(res.final)
print("-" * 70)
if res.link_reply:
    print("\nSUGGESTED FIRST REPLY:")
    print(res.link_reply)
print("\nTIMING:", res.timing_advice)
